src/terraform-module.ts

```diff
@@ -382,35 +382,52 @@ export class TerraformModule {
   public getReleaseType(): ReleaseType | null {
     // If we have commits, analyze them for release type
     if (this.hasDirectChanges()) {
-      const { majorKeywords, minorKeywords } = config;
-      let computedReleaseType: ReleaseType = RELEASE_TYPE.PATCH;
+      const { majorKeywords, minorKeywords, patchKeywords, defaultSemverLevel } = config;
+      let computedReleaseType: ReleaseType | null = null;
+      let anyKeywordMatched = false;
 
       // Analyze each commit message and determine highest release type
       for (const message of this.commitMessages) {
         const messageCleaned = message.toLowerCase().trim();
 
         // Determine release type from current message
-        let currentReleaseType: ReleaseType = RELEASE_TYPE.PATCH;
+        let currentReleaseType: ReleaseType | null = null;
         if (majorKeywords.some((keyword) => messageCleaned.includes(keyword.toLowerCase()))) {
           currentReleaseType = RELEASE_TYPE.MAJOR;
+          anyKeywordMatched = true;
         } else if (minorKeywords.some((keyword) => messageCleaned.includes(keyword.toLowerCase()))) {
           currentReleaseType = RELEASE_TYPE.MINOR;
+          anyKeywordMatched = true;
+        } else if (patchKeywords.some((keyword) => messageCleaned.includes(keyword.toLowerCase()))) {
+          currentReleaseType = RELEASE_TYPE.PATCH;
+          anyKeywordMatched = true;
         }
 
-        // Determine the next release type considering the previous release type
-        if (currentReleaseType === RELEASE_TYPE.MAJOR || computedReleaseType === RELEASE_TYPE.MAJOR) {
-          computedReleaseType = RELEASE_TYPE.MAJOR;
-        } else if (currentReleaseType === RELEASE_TYPE.MINOR || computedReleaseType === RELEASE_TYPE.MINOR) {
-          computedReleaseType = RELEASE_TYPE.MINOR;
+        // Only update computedReleaseType if a keyword was matched in this commit
+        if (currentReleaseType !== null) {
+          // Determine the next release type considering the previous release type
+          if (currentReleaseType === RELEASE_TYPE.MAJOR || computedReleaseType === RELEASE_TYPE.MAJOR) {
+            computedReleaseType = RELEASE_TYPE.MAJOR;
+          } else if (currentReleaseType === RELEASE_TYPE.MINOR || computedReleaseType === RELEASE_TYPE.MINOR) {
+            computedReleaseType = RELEASE_TYPE.MINOR;
+          } else if (computedReleaseType === null) {
+            // First keyword match, set it
+            computedReleaseType = currentReleaseType;
+          }
```

Comment:
Is anyKeywordMatched necessary? Couldn't we check to see computedReleaseType !== null?